appveyor.yml

```diff
@@ -3,20 +3,20 @@ version: "{branch} {build}"
 image: Visual Studio 2017
 
 environment:
-  JAVA_HOME: C:\jdk11
+  JAVA_HOME: C:\jdk12
   APPVEYOR_CACHE_ENTRY_ZIP_ARGS: "-t7z -m0=lzma -mx=9"
 
 shallow_clone: true
 
 build_script:
   - ps: |
       $client = New-Object net.webclient
-      $client.DownloadFile('http://jdk.java.net/11/', 'C:\Users\appveyor\openjdk11.html')
-      $openJdk11 = cat C:\Users\appveyor\openjdk11.html | where { $_ -match "href.*https://download.java.net.*jdk11.*windows-x64.*zip\`"" } | %{ $_ -replace "^.*https:", "https:" } | %{ $_ -replace ".zip\`".*$", ".zip" }
-      echo "Download boot JDK from: $openJdk11"
-      $client.DownloadFile($openJdk11, 'C:\Users\appveyor\openjdk11.zip')
-      Expand-Archive -Path 'C:\Users\appveyor\openjdk11.zip' -DestinationPath 'C:\Users\appveyor\openjdk11'
-      Copy-Item -Path 'C:\Users\appveyor\openjdk11\*\' -Destination 'C:\jdk11' -Recurse -Force
+      $client.DownloadFile('http://jdk.java.net/12/', 'C:\Users\appveyor\openjdk.html')
+      $openJdk = cat C:\Users\appveyor\openjdk.html | where { $_ -match "href.*https://download.java.net.*jdk[1-9][0-9].*windows-x64.*zip\`"" } | %{ $_ -replace "^.*https:", "https:" } | %{ $_ -replace ".zip\`".*$", ".zip" }
+      echo "Download boot JDK from: $openJdk"
+      $client.DownloadFile($openJdk, 'C:\Users\appveyor\openjdk.zip')
+      Expand-Archive -Path 'C:\Users\appveyor\openjdk.zip' -DestinationPath 'C:\Users\appveyor\openjdk'
+      Copy-Item -Path 'C:\Users\appveyor\openjdk\*\' -Destination 'C:\jdk12' -Recurse -Force
```

Comment:
nit: Is it possible to replace the string `C:\jdk12` with `$JAVA_HOME`.

Reply:
It should be possible as long as the `environment` block is executed before the `build_script` block (it's easy enough to check). If so, I'll make that change and then merge it (else I'll merge it as-is).